drivers/include/mtd_emulated.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2023 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mtd
+ * @{
+ * @brief       MTD device that is emulated in RAM for test purposes
+ *
+ * Helpers for using emulated MTDs.
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+#ifndef MTD_EMULATED_H
+#define MTD_EMULATED_H
+
+#include <stdbool.h>
+
+#include "board.h"
+#include "macros/utils.h"
+#include "mtd.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Macro to define an emulated MTD
+ *
+ * This macro creates a MTD device that is emulated in RAM. For example, using
+ * ```
+ * MTD_EMULATED_DEV(0, 16, 4, 64)
+ * ```
+ * creates the emulated MTD device `mtd_emulated_dev0` with 16 sectors, 4 pages
+ * per sector and a page size of 64 bytes. It also defines it as a generic
+ * MTD `mtd2`.
+ *
+ * To use the emulated MTD with the `mtd_default` module, for example, the
+ * `MTD_2` symbol must also be defined before the `mtd_default.h` header file
+ * is included.
```

Comment:
Why is there a reference to MTD_2 and mtd2 here ? 
This macro should create a `mtd_emulated_t` type

Reply:
With the changes in commit https://github.com/RIOT-OS/RIOT/pull/19443/commits/a03d4163a15e845e750e2108e805de9470eccb1b it is no longer necessary to define the emulated MTD also as `MTD_x` macro to get it working with `mtd_default_get_dev`. Therefore, I removed the macro `MTD_EMULATED_DEV_GENERIC` and all the documentation regarding the definition of a generic `mtdx` variable and the `MTD_x` macro.